plugins/logs/plugin_test.go

```diff
@@ -482,6 +482,106 @@ func TestPluginRequeBufferPreserved(t *testing.T) {
 	}
 }
 
+func TestPluginRateLimitInt(t *testing.T) {
+	ctx := context.Background()
+
+	numDecisions := 1 // 1 decision per second
+
+	fixture := newTestFixture(t, func(c *Config) {
+		limit := float64(numDecisions)
+		c.Reporting.MaxDecisionsPerSecond = &limit
+	}, func(c *Config) {
+		limit := int64(300)
+		c.Reporting.UploadSizeLimitBytes = &limit
+	})
+	defer fixture.server.stop()
+
+	var input interface{} = map[string]interface{}{"method": "GET"}
+	var result interface{} = false
+
+	_ = fixture.plugin.Log(ctx, logServerInfo("abc", input, result))
+	_ = fixture.plugin.Log(ctx, logServerInfo("def", input, result))
+
+	_ = fixture.plugin.Log(ctx, logServerInfo("ghi", input, result)) // should not be added to buffer as rate limit exceeded
+
+	if fixture.plugin.buffer.Len() != 1 {
+		t.Fatalf("Expected buffer length of 1 but got %v", fixture.plugin.buffer.Len())
+	}
+
+	<-time.After(1 * time.Second)
```

Comment:
Nit: `time.Sleep` is good enough here, I think

Reply:
Done